### PR TITLE
Add extension point for handling access denied.

### PR DESCRIPTION
**Pull request by adammathys**
Makes overriding desired behaviour when handling unauthorized requests much easier. It can be done on a per-controller basis and defined in outside extensions easily.

```ruby
Spree::BaseController.unauthorized_redirect = -> do
  if try_spree_current_user
    flash[:error] = Spree.t(:authorization_failure)
    redirect_to spree.unauthorzied_path
  else
    store_location
    redirect_to spree.login_path
  end
end
```

Alternative to #43.

### CODE REVIEW

**Comment by jhawthorn** [comment]
:+1:

**Comment by Senjai** [comment]
:+1:

**Comment by magnusvk** [comment]
:+1:

**Comment by forkata** [comment]
:+1:

File: core/lib/spree/core/controller_helpers/auth.rb [hunk 1]
```diff
@@ -8,8 +8,14 @@ module Auth
           before_filter :set_guest_token
           helper_method :try_spree_current_user
 
-          rescue_from CanCan::AccessDenied do |exception|
-            redirect_unauthorized_access
+          class_attribute :unauthorized_redirect
```
Comment (gmacdougall) [inline core/lib/spree/core/controller_helpers/auth.rb]: Can you provide some documentation for this?

**Pull request merged**